File: files/CH-138.R
library(tidyverse)
library(readxl)

path = "files/CH-138 Periodic Sales Summary.xlsx"
input = read_excel(path, range = "C2:E27")
test  = read_excel(path, range = "G2:J6")

result = input %>%
  mutate(Month = month(Date),
         day = day(Date), 
         decade_days = paste0("P",ifelse(ceiling(day / 10) == 4, 3, ceiling(day/10)))) %>%
  summarise(`Total Qty` = sum(Qty, na.rm = TRUE), .by = c(decade_days, Month)) %>%
  pivot_wider(names_from = decade_days, values_from = `Total Qty`, values_fill = 0) %>%
  select(Month, P1, P2, P3)

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE

File: files/100-199/138/CH-138.py
import pandas as pd

path = "CH-138 Periodic Sales Summary.xlsx"
input = pd.read_excel(path, usecols="C:E", skiprows=1, nrows=26)
test = pd.read_excel(path, usecols="G:J", skiprows=1, nrows = 4 )

input['Month'] = input['Date'].dt.month
input['decade_days'] = 'P' + (((input['Date'].dt.day - 1) // 10 + 1).clip(upper=3)).astype(str)

result = input.pivot_table(index = "Month", columns = "decade_days", values = "Qty", aggfunc = "sum").reset_index()
result = result.fillna(0).astype("int64")
result.columns.name = None

print(test.equals(result)) # True